src/Microsoft.ML.Data/DataLoadSave/Text/TextLoader.cs

```diff
@@ -1414,6 +1426,8 @@ private TextLoader(IHost host, ModelLoadContext ctx)
             if (ctx.Header.ModelVerWritten >= 0x0001000D)
             {
                 _escapeChar = ctx.Reader.ReadChar();
+                _decimalMarker = ctx.Reader.ReadChar();
+                host.CheckDecode(_decimalMarker == '.' || _decimalMarker == ',');
             }
             else
```

Comment:
In the "else" case here, you'd also like to initialize the _decimalMarker = Defaults.DecimalMarker;